### PR TITLE
[X86][Driver] Enable feature ndd for -mapxf

### PR DESCRIPTION
**Pull request by KanRobert**

### CODE REVIEW

File: clang/test/Preprocessor/x86_target_features.c [hunk 1]
```diff
@@ -803,10 +803,10 @@
...
 // RUN: %clang -target x86_64-unknown-unknown -march=x86-64 -mapx-features=ccmp -x c -E -dM -o - %s | FileCheck --check-prefix=CCMP %s
 // RUN: %clang -target x86_64-unknown-unknown -march=x86-64 -mapx-features=cf -x c -E -dM -o - %s | FileCheck --check-prefix=CF %s
-// RUN: %clang -target x86_64-unknown-unknown -march=x86-64 -mapxf -x c -E -dM -o - %s | FileCheck --check-prefixes=EGPR,PUSH2POP2,PPX %s
+// RUN: %clang -target x86_64-unknown-unknown -march=x86-64 -mapxf -x c -E -dM -o - %s | FileCheck --check-prefixes=EGPR,PUSH2POP2,PPX,NDD %s
+// CCMP: #define __CCMP__ 1
+// CF: #define __CF__ 1
 // EGPR: #define __EGPR__ 1
+// NDD: #define __NDD__ 1
 // PPX: #define __PPX__ 1
 // PUSH2POP2: #define __PUSH2POP2__ 1
```
Comment (KanRobert) [inline clang/test/Preprocessor/x86_target_features.c:812]: I reorder the check b/c the macro is sorted by alphabetical order.

Comment (phoebewang) [reply clang/test/Preprocessor/x86_target_features.c:812]: Why it wasn't failed before?

Comment (KanRobert) [reply clang/test/Preprocessor/x86_target_features.c:812]: B/c we only checked EGPR, PPX, PUSH2POP2 for `-mapxf`. These three `#define`  in this file was already in alphabetical order.

Comment (phoebewang) [reply clang/test/Preprocessor/x86_target_features.c:812]: Doesn't line 803 check for NDD? If the feature was enabled before, the check of NDD macro should fail.

Comment (KanRobert) [reply clang/test/Preprocessor/x86_target_features.c:812]: Line 803 checks for NDD, but it doesn't matter b/c it does not check other macros. The order for the macros matters only when we need to check several macros in one RUN line.

**Review by phoebewang — APPROVED** [review]
LGTM.

**Comment by KanRobert** [comment]
Thanks! @phoebewang

**Pull request merged**